src/flint/fmpq_abs_series.jl

```diff
@@ -116,6 +116,13 @@ end
 
 characteristic(::QQAbsPowerSeriesRing) = 0
 
+function set_precision!(z::QQAbsPowerSeriesRingElem, k::Int)
+  k < 0 && throw(DomainError(k, "Precision must be non-negative"))
+  z = truncate!(z, k)
+  z.prec = k
```

Comment:
You currently set `z.prec = k` twice: once here in `set_precision!` and once in `truncate!`. Is this intended?

Reply:
`truncate!` does not set the precision if `precision(z) <= k` from the start. `set_precision!` changes the precision "in both directions".
(There is of course room for debate whether this is the expected behaviour.)

Reply:
ah ok. thanks for the explanantion